app/public/js/common/utilsService.js

```diff
@@ -237,6 +238,48 @@ app.factory('utilsService', function(
 
     };
 
+    /**
```

Comment:
I want to understand better what you are trying to achieve because the new code below seems like not necessary.